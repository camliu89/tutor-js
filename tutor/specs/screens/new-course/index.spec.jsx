import User from '../../../src/models/user';
import ModalManager from '../../../src/components/modal-manager';
import NewCourse from '../../../src/screens/new-course';

jest.mock('../../../src/models/loader');
jest.mock('../../../src/models/user', () => ({
  terms_signatures_needed: false,
  isCollegeTeacher: true,
}));

describe('NewCourse wrapper', function() {

  it('requires terms', () => {
    User.terms_signatures_needed = true;
    const wrapper = shallow(<ModalManager><NewCourse /></ModalManager>).dive().dive().dive();
    expect(wrapper).toHaveRendered('TermsModal');
  });

});
