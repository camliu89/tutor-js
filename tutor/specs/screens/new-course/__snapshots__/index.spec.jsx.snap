// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCourse wrapper renders and matches snapshot 1`] = `
<C
  path="/"
>
  <NewCourse />
</C>
`;
